File: TravellerCreator/ClientApp/src/app/sector-creator/models/interfaces/planet.ts
import {TradeCode} from "../enums/trade-code";
import {Base} from "../enums/base";
import {TravelCode} from "../enums/travel-code";
import {PlanetType} from "../enums/planet-type";

export interface IPlanet {
  name: string;
  size: number;
  atmosphere: number;
  temperature: number;
  hydrographics: number;
  population: number;
  government: number;
  lawLevel: number;
  starport: number;
  techLevel: number;
  tradeCodes: TradeCode[];
  bases: Base[];
  travelCode: TravelCode;
  planetType: PlanetType;
  moons: IPlanet[];
}

File: TravellerCreatorModels/Interfaces/Planet.cs
using TravellerCreatorModels.Enums;

namespace TravellerCreatorModels.Interfaces;

public interface IPlanet
{
    string Name { get; set; }
    int Size { get; set; }
    int Atmosphere { get; set; }
    int Temperature { get; set; }
    int Hydrographics { get; set; }
    int Population { get; set; }
    int Government { get; set; }
    int LawLevel { get; set; }
    char Starport { get; set; }
    int TechLevel { get; set; }
    List<TradeCode> TradeCodes { get; set; }
    List<Base> Bases { get; set; }
    TravelCode TravelCode { get; set; }
    public PlanetType PlanetType { get; set; }
    public List<IPlanet> Satellites { get; set; }
}